File: Python3/Subarray Sums Divisible by K.py
# 328ms, 96.72%; 17.8MB, 10.00%
class Solution:
    def subarraysDivByK(self, A: list, K: int) -> int:
        nums = [0] * K
        res, _sum, nums[0] = 0, 0, 1
        for a in A:
            _sum = (_sum + a) % K
            res += nums[_sum]
            nums[_sum] += 1
        return res

File: Cpp/Subarray Sums Divisible by K.cpp
// 56ms, 87.08%; 12.3MB, 75.00%
class Solution
{
public:
    int subarraysDivByK(const vector<int> &A, int K)
    {
        int *nums = new int[K] {1};
        int res = 0, sum = 0;
        for (auto a : A) {
            sum = (sum + a) % K;
            if (sum < 0) sum += K;
            res += nums[sum];
            nums[sum] += 1;
        }
        delete[] nums;
        return res;
    }
};
